models/Document.php

```diff
@@ -262,18 +264,31 @@ public static function getById($id, $force = false)
                 return null;
             }
 
-            $className = 'Pimcore\\Model\\Document\\' . ucfirst($document->getType());
+            // check for if custom types are enabled. Will be removed in Pimcore 11
+            $config = Config::getSystemConfiguration('documents');
 
-            // this is the fallback for custom document types using prefixes
-            // so we need to check if the class exists first
-            if (!Tool::classExists($className)) {
-                $oldStyleClass = 'Document_' . ucfirst($document->getType());
-                if (Tool::classExists($oldStyleClass)) {
-                    $className = $oldStyleClass;
+            if($config['type_definitions_loader_enabled']) {
+                // Getting Typeloader from container
+                $loader = \Pimcore::getContainer()->get(TypeLoader::class);
+                $newDocument = $loader->build($document->getType());
+            } else {
+                /**
+                 * @deprecated since Pimcore 10.6 and will be removed in Pimcore 11. Use type_definitions instead
```

Comment:
This should raise a deprecation message, right?

Reply:
Good point, but how would I raise a deprecation message just for the else part? Where would this message be shown?